### PR TITLE
Move all smartwatch layers to a unified repository.

### PR DESCRIPTION
**Pull request by MagneFire**
Before being merged this depends on https://github.com/MagneFire/meta-smartwatch being upstreamed too.

Instead of adding each layer from `meta-smartwatch` manually use some `find` magic to extract each layer.

This implements part of https://github.com/AsteroidOS/asteroid/issues/194.

### CODE REVIEW

File: prepare-build.sh [hunk 1]
```diff
@@ -150,26 +133,15 @@ BBLAYERS = " \
...
-  ${SRCDIR}/meta-smelt-hybris \
-  ${SRCDIR}/meta-mtk6580-hybris \
-  ${SRCDIR}/meta-mooneye-hybris \
-  ${SRCDIR}/meta-narwhal-hybris \
-  ${SRCDIR}/meta-swift-hybris \
-  ${SRCDIR}/meta-wren-hybris \
-  "' > build/conf/bblayers.conf
+  ${SRCDIR}/meta-openembedded/meta-filesystems \' > build/conf/bblayers.conf
+
+        # Parse all to an array, find layers in meta-smartwatch, exclude root directory, remove most src/, sort alphabetically.
```
Comment (FlorentRevest) [inline prepare-build.sh:138]: The wording is a bit weird. I would reword as:

# Find all layers under src/meta-asteroid, remove the src/ prefix, sorted alphabetically, and store it in an array

**Review by FlorentRevest — APPROVED** [review]
Modulo the comment, I like it, great idea :)

**Comment by FlorentRevest** [comment]
Also I really like https://github.com/MagneFire/meta-smartwatch The only comment would be: why is there a top-level LICENSE file when all the sublayers also have a LICENSE ? It effectively doesn't mean anything. I would either remove all the meta-*-hybris/LICENSE or the top level one

**Comment by MagneFire** [comment]
> Modulo the comment, I like it, great idea :) 

Thanks for the feedback 😊
Are you giving yourself a complement now? 😜 (Given that you were the one that came up with this idea)

I'll try to merge this all this weekend then.

**Comment by FlorentRevest** [comment]
> Are you giving yourself a complement now? :stuck_out_tongue_winking_eye: (Given that you were the one that came up with this idea)

Oh, really ? I must be smarter than I thought then! I have no memory of this ahah

**Comment by LecrisUT** [comment]
Planning for the CI we have 2 options:
- Trigger a build by changes to a specific folders and commit comment like `[Rebuild:smatwatch]`
- Have `meta-smarwatch` have gitmodules of each one, and trigger the builds on the relevant repositories only

In the first option, it is possible to do in TeamCity, but I don't know how portable that configuration is to other CIs. In the second option, it is quite a pain to keep track of the updates to all submodules. Some versioning depending on the common feature list implemented to each device can help with that though.

I personally am ok with either solution, it's just a matter of preference on the organization. I have some comments I'll put in #190 about other organizational changes needed.

**Comment by MagneFire** [comment]
> Planning for the CI we have 2 options:
> - Trigger a build by changes to a specific folders and commit comment like `[Rebuild:smatwatch]`
> - Have `meta-smarwatch` have gitmodules of each one, and trigger the builds on the relevant repositories only
> 
> In the first option, it is possible to do in TeamCity, but I don't know how portable that configuration is to other CIs. In the second option, it is quite a pain to keep track of the updates to all submodules. Some versioning depending on the common feature list implemented to each device can help with that though.
> 
> I personally am ok with either solution, it's just a matter of preference on the organization. I have some comments I'll put in #190 about other organizational changes needed.


The current proposed solution is to have all repositories merged into a single repository. Archiving all _meta-*-hybris_ repositories. In my opinion having git submodules isn't a great solution since you still have the separation, which makes maintenance still the issue, since pushing a change relevant for all watches still requires pushing to the individual repositories.

I agree on portability of your first option, could we also trigger CI builds based on the first word of the commit title? In this new structure most/all commits start with the codename of the watch.
It's not ideal though since we also refer to mtk6580 which are the harmony watches (some kind of mapping would be required).

**Comment by LecrisUT** [comment]
Ok, then I will implement:
- A trigger according to the subfolder in `meta-smartwatch`, i.e. `meta-smartwatch/meta-$MACHINE`. This will trigger regular builds
- A trigger according to any change in common components, excluding the asteroid apps, e.g. `meta-qt`, with a delay in case many changes accumulate
- A few triggers according to commit comment:
  - `[Rebuild]`: rebuilds all without cache
  - `[Rebuild:watch1,watch2,...]`: rebuilds specific watches without cache
  - `[watch1,watch2,...]`: builds for a specific watch
  - Other freeform styles to be discussed
- A timed trigger to rebuild all watches from scratch, compare the caches for any conflicts and upload a fresh set of cache

As for the missing asteroid apps, those will be updated either via the last timed trigger, or over the air, as we can also upload the artifacts to Github and opkg repository. App store also possible ;)

As for the portability of the configurations, I am not sure, but at the very least we can document the required behavior and hope that who works on migrating to other CI can also implement this.
